### ISSUE TITLE
Unhandled exceptions in DashboardController

### ISSUE THREAD

**Issue by DixonDs** [OrleansContrib/OrleansDashboard#23]
Getting this from logs:
```
[2016-12-05 20:06:07.985 GMT     2	ERROR  	100102	Silo	XXX.XXX.XXX.XXX:11111]	
!!!!!!!!!! Silo caught an UnobservedException with context==null.	
Exc level 0: System.TimeoutException: Response did not arrive on time in 00:01:00 for message: Request SXXX.XXX.XXX.XXX:11111:218664204ProviderManagerSystemTarget@S00000013->SXXX.XXX.XXX.XXX:11111:218664204*grn/FFE8F7B3/00000000@d1409a8e #4038: global::OrleansDashboard.IDashboardGrain:GetCounters(). Target History is: <SXXX.XXX.XXX.XXX:11111:218664204:*grn/FFE8F7B3/00000000:@d1409a8e>.
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
at OrleansDashboard.DashboardController.<>c__DisplayClass11_0.<<GetDashboardCounters>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
at OrleansDashboard.DashboardController.<GetDashboardCounters>d__11.MoveNext()
```

Wouldn't it be better to catch and log exceptions in `DashboardController` without passing them through as unobserved exceptions?

**Comment by richorama** [comment]
I think this has already been fixed by this commit: https://github.com/OrleansContrib/OrleansDashboard/commit/29f2e81632c36063d2960538984b0ed3129db9e5

**Comment by DixonDs** [comment]
That commit was included in 1.3.1, right? That is the version we use and see this issue. I think it could be reproduced by adding `Task.Delay` to `DashboardGrain.GetCounters()`. I will try to do it once I get a chance

**Comment by richorama** [comment]
yes, it should be in 1.3.1. I'll take a look later today.
